### DISCUSSION TITLE
Stop Motion detect on selected camera when CPU % is high

### DISCUSSION THREAD

**Discussion by 18923** [blakeblackshear/frigate#3506] [Ideas]
I am running Frigate with 4 low res cameras on an old i5 laptop (without Coral), it's running surprising well. CPU is usually around 55%  however on windy days when there is lots of motion from leaves and shadows (the cameras are all outside) the CPU maxes out above 90% and lots of events are missed.
Of the 4 cameras only 1 is super important that it keeps going so during these conditions I manually stop motion detection on the other 3 cameras.
My thought are to set a switch on the config file to automatically stop detect on a per camera basis as the CPU usage goes up.
For example CPU 70% stop Camera1, CPU 80% stop Camera2, CPU 90% stop Camera3

Might be useful until Corals become available AND affordable.

**Comment by NickM-27** [comment]
I don't think this is something that we'd want to have in the config as we don't have anything watching CPU usage, and this can easily be automated through the HomeAssistant integration or other automation platforms using MQTT. The current integration beta has motion detection switches so this could be done through that no problem.

**Reply by 18923** [reply]
That was quick, thank you. I will look into MQTT.

**Reply by NickM-27** [reply]
this feature is currently only in the beta, you can see it in the docs here: https://deploy-preview-2829--frigate-docs.netlify.app/integrations/mqtt#frigatecamera_namemotionset

**Reply by 18923** [reply]
I assume you mean this... I do not understand the meaning in (NOTE: Turning off motion detection will fail if detection is not disabled.)
frigate/<camera_name>/motion/set
Topic to turn motion detection for a camera on and off. Expected values are ON and OFF. **NOTE: Turning off motion detection will fail if detection is not disabled.**

**Reply by NickM-27** [reply]
If you have object detection enabled and try to turn off motion detection it won't work. You must turn off object detection first. 

It occurred to me that you probably don't need to turn off motion detection and object detection turned off won't hit your CPU very hard, that's already supported in the stable version today

**Reply by 18923** [reply]
I am using the latest Beta.... and was sure that Toggling detect off in the Camera GUI lowered the CPU %
Will try again later.

**Reply by NickM-27** [reply]
> and was sure that Toggling detect off in the Camera GUI lowered the CPU %

Yes it should lower by quite a bit. 

Motion detection could be turned off to make them go even lower but that's probably not necessary (is what I was saying)

**Reply by 18923** [reply]
I was assuming that "Detect off" was like turning Motion detection off, sounds like that is not the case. I will experiment with it a little more next week.

**Reply by NickM-27** [reply]
They are separate, but to understand it requires more knowledge of how frigate works.

Frigate uses motion that is detected and runs object detection in those areas. So naturally when there is a lot of motion there is a lot of object detection queries which is hard on CPU detectors. Turning off object detection resolves this.

In 0.10+ it was made so that motion detection was a separate thing that was not turned off. This is because recordings can be set to only keep segments that have motion, so disabling motion detection would disable 24/7 recording and some users wanted to keep the segments with motion even when object detection was disabled.

In 0.11+ we have added an option to toggle motion detection. This is just an additional toggle that will reduce CPU a bit further but not nearly as much as the detection switch (in CPU detector case)

**Reply by 18923** [reply]
Got it, thank you for the explanation, have a great weekend.
